test_pull_requests

```diff
@@ -152,6 +178,97 @@ def submitted_tests_for_branch(branch)
   $submitted_tests[branch] ? $submitted_tests[branch] : $submitted_tests['*']
 end
 
+# Pulls out as much metadata from the bot comment as it can, including
+# state (AKA content), Jenkins build URL, base repo commit ID, and the
+# extended tests metadata
+def extract_bot_comment_fields(bot_comment_body, settings=nil)
+  test_prefix = nil
+  content = nil
+  build_url = nil
+  base_commit = nil
+  extended_tests = nil
+  state = nil
+  build_url_pattern = nil
+  if settings.nil?
```

Comment:
It might be good to break this out as a separate function, but I like having these regexes inline. They're not reused, and they're tightly coupled to this particular task.